src/mechanism.jl

```diff
@@ -66,4 +77,23 @@ function benefit(players, payoff, reward)
   payoff
 end
 
+function tax_non_grand_coalition(country, grand_coalition; tax_rate = 0.05)
+  grand_coalition_names = Set([prov.name for prov in grand_coalition])
+
+  taxed_country = []
+
+  for prov in country
+      # ∉ is symbol for "not in" one of the only clean ways to do this in julia
+      if prov.name ∉ grand_coalition_names
+          tax = prov.money * tax_rate
+          remaining_money = prov.money - tax
+          push!(taxed_country, Province(prov.name, remaining_money))
+      else
+          push!(taxed_country, prov)
+      end
+  end
+
+  return taxed_country
+end
```

Comment:
I don't think this feature does exactly what we want it to do. The grand coalition is always formed of all members, by definition, so this would essentially just be taxing all of them.

This feature was intended to go through all the possible coalitions, and then for a specific coalition that is not the grand coalition, to tax their combined budget by some percentage and then compute their new utility. The idea here is to make it harder for people to leave the grand coalition and thus possibly make the core exist.

Reply:
Yeah when I went to test it I realized that, I changed it to take in a grand coalition as an argument. I'm also having trouble rebasing now as my branch had some conflict with Daniel's, so a lot more changes are showing up then what I actually did.

Reply:
I just rebased it for you so the Git history is good now.

Then we should have a function that will compute the utility for each of these coalitions now that they've been taxed.

Reply:
I just updated my function and the manner in which I test it.